test/integration/targets/bigip_ssl_key_cert/tasks/main.yaml

```diff
@@ -0,0 +1,89 @@
+---
+
+- name: Import a key
+  bigip_ssl_key_cert:
+    key_content: "{{ lookup('file', 'cert1.key') }}"
+    key_name: cert1
+    state: present
+  register: result
+
+- name: Assert Import a key
+  assert:
+    that:
+      - result is changed
+
+- name: Delete the key
+  bigip_ssl_key_cert:
+    key_name: cert1
+    state: absent
+  register: result
+
+- name: Assert Delete the key
+  assert:
+    that:
+      - result is changed
+
+- name: Import a cert
+  bigip_ssl_key_cert:
+    cert_content: "{{ lookup('file', 'cert1.crt') }}"
+    cert_name: cert1
+    state: present
+  register: result
+
+- name: Assert Import a cert
+  assert:
+    that:
+      - result is changed
+
+- name: Delete the cert
+  bigip_ssl_key_cert:
+    cert_name: cert1
+    state: absent
+  register: result
+
+- name: Assert Delete the cert
+  assert:
+    that:
+      - result is changed
+
+- name: Import both key and cert
+  bigip_ssl_key_cert:
+    key_content: "{{ lookup('file', 'cert1.key') }}"
+    key_name: cert1
+    cert_content: "{{ lookup('file', 'cert1.crt') }}"
+    cert_name: cert1
+    state: present
+  register: result
+
+- name: Assert Import both key and cert
+  assert:
+    that:
+      - result is changed
+
+- name: Update key and cert
+  bigip_ssl_key_cert:
+    key_content: "{{ lookup('file', 'cert2.key') }}"
+    key_name: cert1
+    cert_content: "{{ lookup('file', 'cert2.crt') }}"
+    cert_name: cert1
+    state: present
+  register: result
+
+- name: Assert Update key and cert
+  assert:
+    that:
+      - result is changed
+
+- name: Delete key and cert
+  bigip_ssl_key_cert:
+    key_content: "{{ lookup('file', 'cert2.key') }}"
+    key_name: cert1
+    cert_content: "{{ lookup('file', 'cert2.crt') }}"
+    cert_name: cert1
+    state: absent
+  register: result
+
+- name: Assert Delete key and cert
+  assert:
+    that:
+      - result is changed
```

Comment:
Where are the idempotency tests, also could use tests to address the issue of changing exiting profiles cert/keys (which was the whole point of this exercise)

Reply:
good points. I'll add those.